src/Orchard/Localization/Services/DefaultLocalizedStringManager.cs

```diff
@@ -118,70 +121,49 @@ public class DefaultLocalizedStringManager : ILocalizedStringManager {
             string corePath = string.Format(CoreLocalizationFilePathFormat, culture);
-            string text = _webSiteFolder.ReadFile(corePath);
-            if (text != null) {
-                _localizationStreamParser.ParseLocalizationStream(text, translations, false);
-                if (!DisableMonitoring) {
-                    Logger.Debug("Monitoring virtual path \"{0}\"", corePath);
-                    context.Monitor(_webSiteFolder.WhenPathChanges(corePath));
-                }
-            }
+            LoadTranslationsFromLanguageFilePath(corePath, translations, context, false);
 
             foreach (var module in _extensionManager.AvailableExtensions()) {
                 if (DefaultExtensionTypes.IsModule(module.ExtensionType)) {
                     string modulePath = string.Format(ModulesLocalizationFilePathFormat, module.VirtualPath, culture);
-                    text = _webSiteFolder.ReadFile(modulePath);
-                    if (text != null) {
-                        _localizationStreamParser.ParseLocalizationStream(text, translations, true);
-
-                        if (!DisableMonitoring) {
-                            Logger.Debug("Monitoring virtual path \"{0}\"", modulePath);
-                            context.Monitor(_webSiteFolder.WhenPathChanges(modulePath));
-                        }
-                    }
+                    LoadTranslationsFromLanguageFilePath(modulePath, translations, context, true);
                 }
             }
 
-            foreach (var theme in _extensionManager.AvailableExtensions()) {
-                if (DefaultExtensionTypes.IsTheme(theme.ExtensionType) && _shellDescriptor.Features.Any(x => x.Name == theme.Id))
-                {
-
+            var currentTheme = _workContextAccessor.GetContext().CurrentTheme;
+            var inactiveThemes = _extensionManager.AvailableExtensions().Where(theme=> theme != currentTheme);
+            foreach (var theme in inactiveThemes) {
+                if (DefaultExtensionTypes.IsTheme(theme.ExtensionType) && _shellDescriptor.Features.Any(x => x.Name == theme.Id)){
                     string themePath = string.Format(ThemesLocalizationFilePathFormat, theme.VirtualPath, culture);
-                    text = _webSiteFolder.ReadFile(themePath);
-                    if (text != null) {
-                        _localizationStreamParser.ParseLocalizationStream(text, translations, true);
-
-                        if (!DisableMonitoring) {
-                            Logger.Debug("Monitoring virtual path \"{0}\"", themePath);
-                            context.Monitor(_webSiteFolder.WhenPathChanges(themePath));
-                        }
-                    }
+                    LoadTranslationsFromLanguageFilePath(themePath, translations, context, true);
                 }
             }
 
+            string currentThemePath = string.Format(ThemesLocalizationFilePathFormat, currentTheme.VirtualPath, culture);
```

Comment:
this would previously be handled the same as an inactive theme above and possibly not take priority